error: expected body of lambda expression

Hi, after applying the fix for #689 I see a different build failure on OS X 10.8 or 10.9.

### How to reproduce ###

1. build ccache 4.0 + the fix for #689 on OS X 10.8 with `CXXFLAGS=-stdlib=libc++` or 10.9.

### Actual behavior ###

```
In file included from unittest/main.cpp:26:
unittest/../src/third_party/doctest.h:6162:72: error: expected body of lambda expression
    auto DOCTEST_FIX_FOR_MACOS_LIBCPP_IOSFWD_STRING_LINK_ERRORS = []() DOCTEST_NOINLINE
                                                                       ^
unittest/../src/third_party/doctest.h:305:26: note: expanded from macro 'DOCTEST_NOINLINE'
#define DOCTEST_NOINLINE __attribute__((noinline))
                         ^
1 error generated.
```

### Expected behavior ###

successful compile

### Environment ###

ccache 4.0

OS X Mavericks v10.9.5 (13F1911)
Xcode v6.2 (6C131e)
Apple LLVM version 6.0 (clang-600.0.57) (based on LLVM 3.5svn)

OS X Mountain Lion v10.8.5 (12F2560)
Xcode v5.1.1 (5B1008)
Apple LLVM version 5.1 (clang-503.0.40) (based on LLVM 3.4svn)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,29 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 #
+#
+# Minimum compiler requirements
+# (Fail gracefully instead of cryptic C++ error messages)
+#
+#
+
+# Clang 3.4 and AppleClang 6.0 do not compile doctest.
+# Gcc 4.8.5 has been the minimum version for quite a while.
+if((CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.5)
+OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8.5)
+OR (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0))
+  message(FATAL_ERROR "The compiler you are using is too old, sorry.\nYou need one listed here: https://ccache.dev/platform-compiler-language-support.html")
+endif()
+
+# All Clang problems / special handling ccache has is because of 3.5.
+# All gcc problems / special handling ccache has is because of 4.8.5.
+if((CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.6)
+OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0))
+  message(WARNING "The compiler you are using is rather old.\nIf anything goes wrong you might be better of with one listed here: https://ccache.dev/platform-compiler-language-support.html")
+endif()
+
+
+#
 # Settings
 #
 include(StandardSettings)

